tests/js/autoAdjustTests.js

```diff
@@ -0,0 +1,92 @@
+    "use strict";
+    
+    fluid.registerNamespace("gpii.tests");
+
+    fluid.defaults("gpii.tests.autoAdjustControls", {
+        gradeNames: ["gpii.adjuster.autoAdjust", "autoInit"],
+        model: {
+            autoAdjust: false
+        }
+    });
+    
+    fluid.defaults("gpii.tests.autoAdjust", {
+        gradeNames: ["fluid.test.testEnvironment", "autoInit"],
+        components: {
+            autoAdjuster: {
+                type: "gpii.tests.autoAdjustControls",
+                container: ".gpii-onOffSwitch-container",
+                options: {
+                    strings: {
+                        label: "Auto-adjust",
+                        description: "Preferences will be automatically adjusted based on background noise and ambient light levels.",
+                    }
+                }
+            },
+            autoAdjustTester: {
+                type: "gpii.tests.autoAdjustTester"
+            }
+        }
+    });
+
+    fluid.defaults("gpii.tests.autoAdjustTester", {
+        gradeNames: ["fluid.test.testCaseHolder", "autoInit"],
+        testOptions: {
+            defaultInputStatus: false,
+            newValue: true
+        },
+        modules: [{
+            name: "Auto-adjust adjuster",
+            expect: 4,
+            tests: [{
+                    name: "Description text",
+                    func: "gpii.tests.autoAdjust.validateText",
+                    args: ["Valid description text.", "{autoAdjuster}.dom.autoAdjustDescription", "{autoAdjuster}.options.strings.description"]
+                }, {
+                    name: "Label text",
+                    func: "gpii.tests.autoAdjust.validateText",
+                    args: ["Valid label text.", "{autoAdjuster}.dom.headingLabel", "{autoAdjuster}.options.strings.label"]
+                }, {
+                    name: "Verify checkbox state (unChecked)",
+                    func: "gpii.tests.verifyCheckboxState",
+                    args: ["The auto adjust option is not checked by default", "{that}.options.testOptions.defaultInputStatus", "{autoAdjuster}.dom.valueCheckbox"]
+                }, {
+                    name: "Validate autoAdjust",
+                    expect: 1,
+                    sequence: [{
+                        func: "{autoAdjuster}.triggerModelChangeOnActivate",
+                        args: ["{autoAdjuster}", "{autoAdjuster}.events.afterRender"]
+                    }, {
+                        name: "Verify checkbox state (checked)",
+                        event: "{autoAdjuster}.events.afterRender",
+                        func: "gpii.tests.verifyCheckboxState",
+                        args: ["The auto adjust option is checked", "{that}.options.testOptions.newValue", "{autoAdjuster}.dom.valueCheckbox"]
+                    }]
+                }]
+        }]
+    });
+
+    gpii.tests.autoAdjust.validateText = function (text, elm, expected) {
+        jqUnit.assertEquals(text, expected, elm.text());
+    };
```

Comment:
It would be a good idea to rename "text" in the arguments list as it's a bit confusing since you are using the "text" method on the elm. The "text" in your arguments is just the test message and should probably be called something like "message" or "testMessage". Your messages are also quite similar, so you might want to consider having most of the message predefined in the and then just add in a value for the differences. Example

```javascript
gpii.tests.autoAdjust.validateText = function (elmName, elm, expected) {
    jqUnit.assertEquals("The text for the '" + elmName +"' should be set correctly", expected, elm.text());
};
```